### PR TITLE
Add a note to call out that Zfinx always follows the "soft-float" ABI

### PR DESCRIPTION
**Pull request by jrtc27**
Closes: #166

### CODE REVIEW

**Review by jim-wilson — APPROVED** [review]

File: riscv-elf.md [hunk 1]
```diff
@@ -319,6 +319,11 @@ operating mode.
 The \*F ABIs require the \*F ISA extension, the \*D ABIs require the \*D ISA
 extension, and the LP64Q ABI requires the Q ISA extension.
 
+> NOTE:
+> This means code targeting the Zfinx extension always uses the ILP32, ILP32E
+> or LP64 integer calling-convention only ABIs as there is no dedicated
```
Comment (luismarques) [inline riscv-elf.md:324]: "integer calling-convention only ABIs" -> "integer only calling-convention ABIs"?

Comment (jrtc27) [reply riscv-elf.md:324]: That does seem like a better way round, but I copied the description used for ILP32 and LP64 above, which have it this way round

Comment (jrtc27) [reply riscv-elf.md:324]: Currently the document is written as "there is an integer calling convention and you can optionally add a floating point calling convention to it", which is a slightly weird way of thinking of it that doesn't quite match what is actually happening

**Review by kito-cheng — APPROVED** [review]
LGTM.

**Comment by kito-cheng** [comment]
But let us wait one more LGTM from LLVM community.

**Comment by asb** [comment]
> But let us wait one more LGTM from LLVM community.

LGTM from the LLVM side.

**Comment by jrtc27** [comment]
Rebased to resolve merge conflict. Ok to merge?

**Pull request merged**